client/src/App.js

```diff
@@ -2,19 +2,40 @@ import React, { Component } from "react";
 import Search from "./Search";
 import Calendar from "./Calendar";
 import Results from "./Results";
+import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
 
 class App extends Component {
   constructor(props) {
     super(props);
+
+    this.state = {
+      term: ""
+    };
+
+    this.onSearchSubmit = this.onSearchSubmit.bind(this);
+  }
+
+  onSearchSubmit(term) {
```

Comment:
Consider using an arrow function instead. Then you don't need to bind the function.

From Google's JS [styleguide](https://google.github.io/styleguide/jsguide.html#features-functions-arrow-functions):
> Never call f.bind(this) or goog.bind(f, this) (and avoid writing const self = this). All of these can be expressed more clearly and less error-prone with an arrow function.

Reply:
Bind is also [slower](https://stackoverflow.com/questions/17638305/why-is-bind-slower-than-a-closure/17638540#17638540) than a clouse

Reply:
The performance difference is minuscule, but I changed it in a new commit